org.eclipse.buildship.core/src/main/java/org/eclipse/buildship/core/internal/workspace/GradleClasspathContainerUpdater.java

```diff
@@ -117,6 +119,21 @@ private void updateClasspathContainer(PersistentModelBuilder persistentModel, IP
         return result.build();
     }
 
+    private boolean isZipArchiveSuffix(String dependencyName) {
+       String dependencyNameLwc = dependencyName.toLowerCase();
+        return dependencyNameLwc.endsWith(".jar") || dependencyNameLwc.endsWith(".rar") || dependencyNameLwc.endsWith(".zip");
+    }
+
+    private boolean isZipArchive(File dependencyFile) {
```

Comment:
❌ I think we should avoid doing extra IO operations here. Filtering based on the extensions should be fine. Is there a non-accidental use case where we cannot expect the proper file extension?

Reply:
@donat in our case we do only have .rar at the moment, so that could be left out. Eventually we should make those extensions configurable to catch additional ones if feasible...

Reply:
That makes sense. We could have a workspace-scoped configuration with a list of accepted extensions. I'm happy to help if you are interested in contributing this feature.

Reply:
That makes sense. We could have a workspace-scoped configuration with a list of accepted extensions. I'm happy to help if you are interested in contributing this feature.